middleware.go

```diff
@@ -0,0 +1,92 @@
+package jwt
+
+import (
+	"github.com/Digital-Insight-Technologies-Ltd/gin-jwt-cognito/models"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-errors/errors"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/google/uuid"
+)
+
+// UserClaimsKey is the key for user claims in our context
+const UserClaimsKey = "userClaims"
+
+// CognitoClaimsMiddleware is a middleware for Gin that extracts user claims from
+// AWS Cognito.
+func CognitoClaimsMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		claims, err := parseClaims(c.GetHeader("Authorization")[7:])
```

Comment:
Validate the presence and format of the Authorization header before slicing it. For example, check if the header starts with "Bearer " and handle cases where it does not. [important]